src/py/flwr/superexec/app.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2020 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Flower superexec app."""
+
+import argparse
+import sys
+from logging import INFO, WARN
+from pathlib import Path
+from typing import Optional, Tuple
+
+import grpc
+
+from flwr.common import EventType, event, log
+from flwr.common.address import parse_address
+from flwr.common.exit_handlers import register_exit_handlers
+from flwr.common.object_ref import load_app, validate
+
+from .exec_grpc import run_superexec_api_grpc
+from .executor import Executor
+
+
+def run_superexec() -> None:
+    """Run Flower SuperExec."""
+    log(INFO, "Starting Flower SuperExec")
+
+    event(EventType.RUN_SUPEREXEC_ENTER)
+
+    args = _parse_args_run_superexec().parse_args()
+
+    # Parse IP address
+    parsed_address = parse_address(args.address)
+    if not parsed_address:
+        sys.exit(f"SuperExec IP address ({args.address}) cannot be parsed.")
+    host, port, is_v6 = parsed_address
+    address = f"[{host}]:{port}" if is_v6 else f"{host}:{port}"
+
+    # Obtain certificates
+    certificates = _try_obtain_certificates(args)
+
+    # Start SuperExec API
+    superexec_server: grpc.Server = run_superexec_api_grpc(
+        address=address,
+        plugin=_get_exec_plugin(args),
+        certificates=certificates,
+    )
+
+    grpc_servers = [superexec_server]
+
+    # Graceful shutdown
+    register_exit_handlers(
+        event_type=EventType.RUN_SUPEREXEC_ENTER,
+        grpc_servers=grpc_servers,
+        bckg_threads=None,
+    )
+
+    # Block
+    while True:
+        superexec_server.wait_for_termination(timeout=1)
```

Comment:
Why do we use a busy loop here?

Reply:
We want the superexec server to run until we shut it down

Reply:
We can use `wait_for_termination` without a `timeout` for that